### PR TITLE
Encoders

### PR DESCRIPTION
**Pull request by LordDarkula**
Port encoders from code-2016

https://app.asana.com/0/219272915179160/256658680119783

### CODE REVIEW

File: frc/src/main/scala/com/lynbrookrobotics/potassium/frc/TalonEncoder.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import com.ctre.CANTalon
+import com.lynbrookrobotics.potassium.{PeriodicSignal, Signal}
+import com.lynbrookrobotics.potassium.units.Ratio
+import squants.{Angle, Dimensionless, Each}
+import squants.motion.AngularVelocity
+import squants.time.Seconds
+
+class TalonEncoder(talon: CANTalon, conversionFactor: Ratio[Angle, Dimensionless]) {
+  def getAngle(): Angle = {
```
Comment (PhilipAxelrod) [inline frc/src/main/scala/com/lynbrookrobotics/potassium/frc/TalonEncoder.scala:11]: Nit: extra parentheses

File: frc/src/main/scala/com/lynbrookrobotics/potassium/frc/TalonEncoder.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    conversionFactor * Each(talon.getPosition()) * 10
+  }
+
```
Comment (PhilipAxelrod) [inline frc/src/main/scala/com/lynbrookrobotics/potassium/frc/TalonEncoder.scala:14]: You might as well rl put in the body of getAngle

File: frc/src/main/scala/com/lynbrookrobotics/potassium/frc/TalonEncoder.scala [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  val angle: Signal[Angle] = Signal {
+    getAngle()
+  }
+
+  val angularVelocity: Signal[AngularVelocity] = Signal {
```
Comment (PhilipAxelrod) [inline frc/src/main/scala/com/lynbrookrobotics/potassium/frc/TalonEncoder.scala:19]: See above comment

File: frc/src/main/scala/com/lynbrookrobotics/potassium/frc/TalonEncoder.scala (hunk 2 above)
Comment (shadaj) [reply frc/src/main/scala/com/lynbrookrobotics/potassium/frc/TalonEncoder.scala:14]: 👍

**Comment by shadaj** [comment]
Instead of adding merge commits, can you please `git rebase master` next time? We want to have just one commit per PR.

Also, since you now have multiple commits, you might want to `git rebase -i master` and squash all commits except the first one.

**Review by shadaj — APPROVED** [review]
Going to merge now, with tests later.

**Pull request merged**